### DISCUSSION TITLE
GUI, meshing and compiled distribution for Capytaine

### DISCUSSION THREAD

**Discussion by RubendeBruin** [capytaine/capytaine#323] [Ideas]
I've been using capytaine occasionally for building 1st order hydrodynamic databases.

For my DAVE project, I've put together a quick GUI to make ship and offshore structure meshes using
- boolean geometry 
- frames
- existing meshes.

[Github link](https://github.com/RubendeBruin/pymeshup)
[Images and executable](https://www.open-ocean.org/pymeshup/)

As it turns out it is quite easy to generate a panel mesh from the resulting models using the isotropic explicit remeshing.

As such it could be a good basis for a basic gui for capytaine.

I've never tried to compile/package capytaine using pyinstaller but if that works as well then we could offer a completely stand-alone version of capytaine with a gui.

Anyways, I was wondering if there is any interest in developing/using a Gui like this.

**Comment by mancellin** [comment]
About GUI: I'm not planning to work on a GUI anytime soon, except maybe for some improvements of the existing 3D visualization tools. A GUI would probably be useful to some users, but I prefer to focus on improving the internal of the solver where my expertise is more valuable. If you'd like to work on such a project, I'd be happy to help you with the technical details and the advertisement.

About PyInstaller: I'm really concerned about the user-friendliness of the installation of Capytaine, especially on Windows. There was a discussion with @izabala123 about a stand-alone installation of Capytaine (to use its command line interface) a long time ago, but I don't think it lead to anything. When I have some time, I should check how PyInstaller is supporting binary extensions to Python code.

**Reply by RubendeBruin** [reply]
Hi @mancellin , thanks for the response. I understand that your focus should be on the math rather than a gui.

I can quickly try to build a pyinstaller package just to see if it works.

**Reply by mancellin** [reply]
Some tries with PyInstaller: https://github.com/capytaine/capytaine/pull/324
